### PR TITLE
Angular update & Added effects

### PR DESCRIPTION
**Pull request by NejcZdovc**
Updated angular to the version 2.1.0-beta.0 and ng2-select2 to version 0.5.1. Because the fix in version 2.1.0-beta.0 solved router problem I could implemented effects.

### CODE REVIEW

**Comment by NejcZdovc** [comment]
Commit renamed

File: src/store/storage/local.ts [hunk 1]
```diff
@@ -32,8 +32,12 @@ export class Local {
         }
 
         return {
-            keymaps: config.keymaps,
-            macros: config.macros,
+            keymaps: {
```
Comment (fjozsef) [inline src/store/storage/local.ts]: Why does it need to be wrapped in the object? I thought we have already talked about this and it worked with arrays perfectly.

File: src/store/reducers/keymap.ts [hunk 2]
```diff
@@ -70,40 +89,51 @@ export default function(state = initialState, action: Action): Keymap[] {
...
+                entities: newState
+            };
+
         default: {
             return state;
         }
     }
 }
 
+export function getKeymapEntities() {
```
Comment (fjozsef) [inline src/store/reducers/keymap.ts]: Add return type

File: src/store/reducers/keymap.ts [hunk 3]
```diff
@@ -70,40 +89,51 @@ export default function(state = initialState, action: Action): Keymap[] {
...
+    return (state$: Observable<AppState>) => state$
+        .select(state => state.keymaps.entities);
+}
+
 export function getKeymap(abbr: string) {
     if (abbr === undefined) {
         return getDefault();
     }
 
     return (state$: Observable<AppState>) => state$
```
Comment (fjozsef) [inline src/store/reducers/keymap.ts]: You could use `getKeymapEntities` here.

File: src/store/reducers/keymap.ts [hunk 4]
```diff
@@ -70,40 +89,51 @@ export default function(state = initialState, action: Action): Keymap[] {
...
-        .select(appState => appState.keymaps)
+        .select(appState => appState.keymaps.entities)
         .map((keymaps: Keymap[]) =>
             keymaps.find((keymap: Keymap) => keymap.abbreviation === abbr)
         );
 }
 
 export function getDefault() {
     return (state$: Observable<AppState>) => state$
```
Comment (fjozsef) [inline src/store/reducers/keymap.ts]: `getKeymapEntities` could be used here, too.

File: src/store/reducers/macro.ts [hunk 5]
```diff
@@ -96,16 +121,24 @@ export default function(state = initialState, action: Action): Macro[] {
...
+                entities: newState
+            };
+
+        default:
             return state;
-        }
     }
 }
 
+export function getMacroEntities() {
```
Comment (fjozsef) [inline src/store/reducers/macro.ts]: Add return type.

File: src/store/reducers/macro.ts [hunk 6]
```diff
@@ -96,16 +121,24 @@ export default function(state = initialState, action: Action): Macro[] {
...
+    return (state$: Observable<AppState>) => state$
+        .select(state => state.macros.entities);
+}
+
 export function getMacro(id: number) {
     if (isNaN(id)) {
         return (state$: Observable<AppState>) => state$
```
Comment (fjozsef) [inline src/store/reducers/macro.ts]: Use `getMacroEntities`

File: src/store/reducers/macro.ts [hunk 7]
```diff
@@ -115,7 +148,7 @@ export function getMacro(id: number) {
             });
     } else {
         return (state$: Observable<AppState>) => state$
```
Comment (fjozsef) [inline src/store/reducers/macro.ts]: Use `getMacroEntities`

File: src/store/reducers/keymap.ts (hunk 2 above)
Comment (NejcZdovc) [inline src/store/reducers/keymap.ts]: Done

File: src/store/reducers/macro.ts (hunk 5 above)
Comment (NejcZdovc) [inline src/store/reducers/macro.ts]: Done

**Pull request merged**